src/services/users/reset-password/reset-password.test.js

```diff
@@ -0,0 +1,39 @@
+const { expect } = require('chai')
+const { faker } = require('@faker-js/faker')
+const generateAccessToken = require('../generate-access-token')
+const {fixtures} = require('../../../../test/utils')
+const resetPassword = require('./index')
+const db = require('../../../db')
+
+
+describe('Reset user password and verify token', () => {
+	let user
+
+	before(async ()=>{
+		 user = await fixtures.insertUser() 
+	})
+
+	it("should reset a user's password", async () => {
+		const token = generateAccessToken(
+			{data: {email: user.email}, 
+				expiry: '1h'})
+
+		await resetPassword({token, newPassword: faker.internet.password()})
+		const {rows} = await db.query(
+			`SELECT * FROM users
+             WHERE email = $1`,[user.email])
+		const updatedUser = rows[0]
+		expect(updatedUser.passwordHash).to.not.equal(user.passwordHash)
+	})
+
+	it('should throw an error if the token is invalid', async () => {
+		try{
```

Comment:
nice one, for this instead of the try catch, we will be using an additional extension of `chai` called `chai-as-promised` which can allow you to check errors thrown from async functions. 

But for that, to show how it's used and since it should have been part of the original setup, ,I will push that part as part of your pr so you can focus on the other reviews 

so your error test will now become. 

```js
return expect(resetPassword(
  token: faker.datatype.uuid(), 
  newPassword: faker.internet.password()}))
.to.be.rejectedWith('Token is expired/invalid.')
```

Reply:
> But for that, to show how it's used and since it should have been part of the original setup, ,I will push that part as part of your pr so you can focus on the other reviews

just did this